homeassistant/components/sensor/yweather.py

```diff
@@ -0,0 +1,172 @@
+"""
+Support for the Yahoo! Weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.yweather/
+"""
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.const import (CONF_PLATFORM, TEMP_CELSIUS,
+                                 CONF_MONITORED_CONDITIONS, STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ["yahooweather==0.3"]
+
+SENSOR_TYPES = {
+    'weather_current': ['Current', None],
+    'weather': ['Condition', None],
+    'temperature': ['Temperature', "temperature"],
+    'temp_min': ['Temperature', "temperature"],
+    'temp_max': ['Temperature', "temperature"],
+    'wind_speed': ['Wind speed', "speed"],
+    'humidity': ['Humidity', "%"],
+    'pressure': ['Pressure', "pressure"],
+    'visibility': ['Visibility', "distance"],
+}
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): "yweather",
+    vol.Required("woeid"): vol.Coerce(str),
+    vol.Optional("forecast"): vol.Coerce(int),
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=[]):
+        [vol.In(SENSOR_TYPES.keys())],
+})
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yahoo! weather sensor."""
+    from yahooweather import UNIT_C, UNIT_F
+
+    unit = hass.config.temperature_unit
+    woeid = config.get("woeid")
```

Comment:
If no woeid given, we could fetch the woeid using the following YQL:

```sql
SELECT woeid FROM geo.places WHERE text="(32.8733367,-117.2274053)"
```

([test in console](http://developer.yahoo.com/yql/console/?q=select%20*%20from%20weather.forecast%20where%20woeid%20in%20(SELECT%20woeid%20FROM%20geo.places%20WHERE%20text%3D%22(32.87336%2C-117.22743)%22)))